lib/Mir.ml

```diff
@@ -96,18 +262,66 @@ type ('e, 's) prog =
   ; prog_path: string }
 [@@deriving sexp]
 
+let pp_input_vars pp_e ppf {input_vars; _} =
+  pp_io_var_block "input_vars" pp_e ppf input_vars
+
+let pp_output_vars pp_e ppf {output_vars; _} =
+  pp_io_var_block "output_vars" pp_e ppf output_vars
+
+let pp_functions_block pp_s ppf {functions_block; _} =
+  pp_block "functions" pp_s ppf functions_block
+
+let pp_prepare_data pp_s ppf {prepare_data; _} =
+  pp_block "prepare_data" pp_s ppf prepare_data
+
+let pp_prepare_params pp_s ppf {prepare_params; _} =
+  pp_block "prepare_params" pp_s ppf prepare_params
+
+let pp_log_prob pp_s ppf {log_prob; _} = pp_block "log_prob" pp_s ppf log_prob
+
+let pp_generate_quantities pp_s ppf {generate_quantities; _} =
+  pp_block "generate_quantities" pp_s ppf generate_quantities
+
+let pp_transform_inits pp_s ppf {transform_inits; _} =
+  pp_block "transform_inits" pp_s ppf transform_inits
+
+(* { functions_block: 's list
+  ; input_vars: 'e io_var list
+  ; prepare_data: 's list (* data & transformed data decls and statements *)
+  ; prepare_params: 's list (* param & tparam decls and statements *)
+  ; log_prob: 's list (*assumes data & params are in scope and ready*)
+  ; generate_quantities: 's list (* assumes data & params ready & in scope*)
+  ; transform_inits: 's list
+  ; output_vars: 'e io_var list
+  ; prog_name: string
+  ; prog_path: string }
+   *)
+let pp_prog pp_e pp_s ppf prog =
+  Fmt.pf ppf "@[<v>@;%a@;%a@;%a@;%a@;%a@;%a@;%a@;%a@]"
```

Comment:
I always wonder if there's some more readable way to format this. For example I think I might find this pattern slightly more digestible if also more repetitive:
https://github.com/stan-dev/stanc3/blob/master/lib/Stan_math_code_gen.ml#L472

Reply:
Agreed; I actually hit this being hard to read when implementing. I'll have a look at alternatives and follow the style in the linked file if I don't have anything better.